src/main/java/laura/TaskList.java

```diff
@@ -0,0 +1,192 @@
+            default -> throw new DecodeException();
+        };
+    }
+
+    /**
+     * Load the Tasks saved from local into the TaskList
+     */
+    public void load() {
+        File file = new File(this.dataPath);
+        try {
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String data = scanner.nextLine();
+                try {
+                    this.list.add(this.decode(data));
+                } catch (LauraException e) {
+                    UI.send(e.getMessage());
+                }
+            }
+        } catch (FileNotFoundException e) {
+            UI.send(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Add a Task to the TaskList
+     * @param task The task to be added
+     */
+    public void add(Task task) {
+        this.list.add(task);
+        this.save();
+        UI.send("Got it! I've added this task:\n" + task);
+    }
+
+    /**
+     * Delete a Task from the TaskList
+     * @param index The index of the Task in the TaskList to be deleted
+     * @throws LauraException If there is no corresponding Task for the index given
+     */
+    public void delete(int index) throws LauraException {
+        if (index > this.list.size() || index < 1) {
+            throw new LauraException("Sorry, that task does not exist!");
+        }
+        Task removed = this.list.remove(index - 1);
+        this.save();
+        UI.send("Noted. I've removed this task:\n"
+                + removed +
+                "\nNow you have " + this.list.size() + " in this list.");
+    }
+
+    /**
+     * Mark a Task from the TaskList
+     * @param index The index of the Task in the TaskList to be marked
+     * @throws LauraException If there is no corresponding Task for the index given
+     */
+    public void mark(int index) throws LauraException {
+        if (index > this.list.size() || index < 1) {
+            throw new LauraException("Sorry, that task does not exist!");
+        }
+        Task curr = this.list.get(index - 1);
+        curr.mark();
+        this.save();
+        UI.send("Nice! I've marked this task as done:\n" + curr);
+    }
+
+    /**
+     * Unmark a Task from the TaskList
+     * @param index The index of the Task in the TaskList to be unmarked
+     * @throws LauraException If there is no corresponding Task for the index given
+     */
+    public void unmark(int index) throws LauraException {
+        if (index > this.list.size() || index < 1) {
+            throw new LauraException("Sorry, that task does not exist!");
+        }
+        Task curr = this.list.get(index - 1);
+        curr.unmark();
+        this.save();
+        UI.send("Ok! I've marked this task as not done:\n" + curr);
+    }
+
+    /**
+     * Filter the current TaskList to only Tasks which has the keyword
+     * @param keyword The keyword to filter by
+     */
+    public void find(String keyword) {
+        ArrayList<Task> filtered = new ArrayList<>();
```

Comment:
Perhaps a more intuitive name here?
```suggestion
        ArrayList<Task> filteredTasks = new ArrayList<>();
```